pysat/tests/test_orbits.py

```diff
@@ -558,22 +559,12 @@ def teardown(self):
 def filter_data(inst):
     """Remove data from instrument, simulating gaps"""
 
-    times = [[pysat.datetime(2009, 1, 1, 10),
-              pysat.datetime(2009, 1, 1, 12)],
-             [pysat.datetime(2009, 1, 1, 4),
-              pysat.datetime(2009, 1, 2, 5, 37)],
-             [pysat.datetime(2009, 1, 1, 1, 37),
-              pysat.datetime(2009, 1, 1, 3, 14)],
-             [pysat.datetime(2009, 1, 1, 15),
-              pysat.datetime(2009, 1, 1, 16)],
-             [pysat.datetime(2009, 1, 1, 22),
-              pysat.datetime(2009, 1, 2, 2)],
-             [pysat.datetime(2009, 1, 13),
-              pysat.datetime(2009, 1, 15)],
-             [pysat.datetime(2009, 1, 20, 1),
-              pysat.datetime(2009, 1, 25, 23)],
-             [pysat.datetime(2009, 1, 25, 23, 30),
-              pysat.datetime(2009, 1, 26, 3)]
+    times = [[dt.datetime(2009, 1, 1, 1, 37), dt.datetime(2009, 1, 1, 3, 14)],
+             [dt.datetime(2009, 1, 1, 10), dt.datetime(2009, 1, 1, 12)],
+             [dt.datetime(2009, 1, 1, 22), dt.datetime(2009, 1, 2, 2)],
+             [dt.datetime(2009, 1, 13), dt.datetime(2009, 1, 15)],
+             [dt.datetime(2009, 1, 20, 1), dt.datetime(2009, 1, 25, 23)],
+             [dt.datetime(2009, 1, 25, 23, 30), dt.datetime(2009, 1, 26, 3)]
```

Comment:
The time gaps here to remove overlap, particularly around the Jan 1 / jan 2 boundary.  Running into fatal error stack overflow issues here in some of the pull requests.  Simplified this to reduce the nested loops in these tests.  Probably a lot more we can do here, but this is working for now.

Errors in:
pysat/tests/test_orbits.py::TestOrbitsGappyData::test_repeat_orbit_calls_asym_multi_day_0_UT_multiple_time_gaps

See:
https://travis-ci.org/github/pysat/pysat/jobs/669972453?utm_medium=notification&utm_source=github_status

Reply:
I took a look at the line listed in the error log and I don't know what is going on there.

Reply:
I wonder if it is related to the fact that the test uses inst.index. Line 410, referenced in the test error log a bunch, is related to attributes.

We may need to review what kind of attribute mapping and other features we have going on and reduce functionality a bit. Outside of this pull.

Reply:
Definitely.  I should also note that the stack overflow error has occurred sporadically as part of this pull, on different tests, but always in the gappy orbits, which is what leads me to suspect the filter_times function.  I can't definitively say that the simplifications above fixed the problem, as sometimes this goes away after updating unrelated parts of the code.  I agree this is outside of the pull.